### PR TITLE
Few fixes to match CTT

### PR DESCRIPTION
**Pull request by nicolas71640**
Here's the status of CTT with this PR : 

<!DOCTYPE html>

<html>
<body>

TestCase | Name | Type
-- | -- | --
TF-1300_1.7.0-1-1 | Secondary Vendor ID | Success
TF-1300_1.7.0-1-2 | Device.RevisionNo | Success
TF-1300_1.7.0-1-3 | ESI Port Type Consistency | Success
TF-1300_1.7.0-1-4 | Number of Offline Dictionaries | Success
TF-1300_1.7.0-1-6 | Device:Name | Success
TF-1300_1.7.0-1-8 | Device:GroupType | Success
TF-1300_1.7.0-1-9 | Device:Mailbox:DataLinkLayer | Success
TF-1300_1.7.0-2-1 | Vendor ID | Success
TF-1300_1.7.0-2-2 | Product Code | Success
TF-1300_1.7.0-2-3 | Revision Number | Success
TF-1300_1.7.0-2-5 | Mailbox Protocols | Success
TF-1300_1.7.0-2-6 | E²PROM Size | Success
TF-1300_1.7.0-2-8 | E²PROM ConfigData | Success
TF-1300_1.7.0-2-10 | E²PROM General Category | Success
TF-1300_1.7.0-2-11 | E²PROM FMMU Categories | Success
TF-1300_1.7.0-2-12 | E²PROM SyncM Categories | Success
TF-1300_1.7.0-3-3 | Number of Fmmu | Success
TF-1300_1.7.0-3-5 | Number of Sm | Success
TF-1300_1.7.0-4-1 | ESI OD Identity Info | Success
TF-1300_1.7.0-5-1 | Vendor:Image16x14 | Success
TF-1300_1.7.0-5-2 | Vendor@UniqueName | Success
TF-1300_1.7.0-5-3 | Group:Image16x14 | Success
TF-1300_1.7.0-5-4 | Device:Image16x14 | Success
TF-1300_1.7.0-5-6 | Device:Type@TcCfgModeSafeOp | Success
TF-1300_1.7.0-5-8 | Device:Profile@Channel | Success
TF-1300_1.7.0-5-9 | Device:Fmmu@OpOnly | Success
TF-1300_1.7.0-5-10 | Device:Sm@OneByteMode | Success
TF-1300_1.7.0-5-11 | Device:Sm@Wachdog | Success
TF-1300_1.7.0-5-12 | Device:Sm@FixedAssignment | Success
TF-1300_1.7.0-5-13 | Device:RxPdo@PdoOrder | Success
TF-1300_1.7.0-5-14 | Device:TxPdo@PdoOrder | Success
TF-1300_1.7.0-5-15 | Device:Mailbox:CoE:Object | Success
TF-1300_1.7.0-5-17 | Device:Mailbox:CoE@DS402Channels | Success
TF-1300_1.7.0-5-24 | Device:Profile:Dictionary:Object:Flags:Transition | Success
TF-1300_1.7.0-5-25 | Device:Profile:Dictionary:DataType:SubItem:DefaultString | Success
TF-1300_1.7.0-5-26 | Device:Profile:Dictionary:DataType:SubItem:DefaultData | Success
TF-1300_1.7.0-5-27 | Device:Profile:Dictionary:DataType:SubItem:MinValue | Success
TF-1300_1.7.0-5-28 | Device:Profile:Dictionary:DataType:SubItem:MaxValue | Success
TF-1300_1.7.0-5-29 | Device:Profile:Dictionary:DataType:SubItem:DefaultValue | Success
TF-1300_1.7.0-5-30 | Device:Profile:Dictionary:DataType:SubItem:Xml | Success
TF-1300_1.7.0-6-1 | Device:Type@DownloadModuleList | Success
TF-1300_1.7.0-6-2 | Device:AlternativeType | Success
TF-1300_1.7.0-6-4 | Device:Mailbox@RealTimeMode | Success
TF-1300_1.7.0-6-6 | Device:Profile:Dictionary:DataType:SubItem:DisplayName | Success
TF-1300_1.7.0-6-7 | Device:Profile:Dictionary:DataTypes:DataType:Index | Success
TF-1300_1.7.0-6-8 | Device:InfoType:DeviceFeature | Success
TF-1300_1.7.0-7-1 | Check Attributes | Success
TF-1300_1.7.0-7-2 | Check Index | Success
TF-1300_1.7.0-7-3 | Check SM assignment | Success
TF-1300_1.7.0-7-4 | Check Pdo entry Data Types | Success
TF-1300_1.7.0-7-5 | Check Pdo Default Size | Success
TF-1300_1.7.0-7-5 | Check Pdo Default Size | Success
TF-1300_1.7.0-7-6 | Check Pdo Exclude | Success
TF-1300_1.7.0-8-1 | Check Attributes | Success
TF-1300_1.7.0-8-2 | Check Index | Success
TF-1300_1.7.0-8-3 | Check SM assignment | Success
TF-1300_1.7.0-8-4 | Check Pdo entry Data Types | Success
TF-1300_1.7.0-8-5 | Check Pdo Default Size | Success
TF-1300_1.7.0-8-5 | Check Pdo Default Size | Success
TF-1300_1.7.0-8-6 | Check Pdo Exclude | Success
TF-1300_1.7.0-9-1 | ESC supported FMMUs | Success
TF-1300_1.7.0-9-2 | FMMU Names | Success
TF-1300_1.7.0-9-3 | FMMU 'MBoxState' | Warning
TF-1300_1.7.0-9-3 | FMMU 'MBoxState' | Success
TF-1300_1.7.0-9-4 | FMMU 'Inputs' | Success
TF-1300_1.7.0-9-5 | FMMU 'Outputs' | Success
TF-1300_1.7.0-9-6 | Process data length test | Success
TF-1300_1.7.0-10-1 | All DataType Names | Success
TF-1300_1.7.0-10-2 | Check Access Flag | Success
TF-1300_1.7.0-10-3 | BaseDataTypes (fixed length) | Success
TF-1300_1.7.0-10-6 | Array/Record DataTypes | Success
TF-1300_1.7.0-11-1 | General Object Attributes | Success
TF-1300_1.7.0-11-2 | Object DataType and BitSize | Success
TF-1300_1.7.0-11-4 | Object Info (Array/Record) | Success
TF-1300_1.7.0-12-1 | Text | Success
TF-1300_1.7.0-12-2 | Sm Order | Success
TF-1300_1.7.0-12-3 | SyncMan start addresses | Success
TF-1300_1.7.0-12-4 | SyncMan default size | Success
TF-1300_1.7.0-12-5 | SM@Enable | Success
TF-1100_1.4.0-1-1 | MBXERR_UNSUPPORTEDPROTOCOL1 | Success
TF-1100_1.4.0-1-2 | MBXERR_UNSUPPORTEDPROTOCOL2 | Success
TF-1100_1.4.0-1-3 | MBXERR_INVALIDSIZE | Success
TF-1100_1.4.0-1-4 | MBXERR_INVALIDHEADER | Success
TF-1100_1.4.0-2-1 | Incremented Master Mailbox Counter | Success
TF-1100_1.4.0-2-2 | Invalid Master Mailbox Counter | Error
TF-1100_1.4.0-2-2 | Invalid Master Mailbox Counter | Success
TF-1100_1.4.0-2-3 | Mailbox Write Service Repeat | Success
TF-1100_1.4.0-2-4 | Mailbox Read Service Repeat 1 | Error
TF-1100_1.4.0-2-4 | Mailbox Read Service Repeat 1 | Error
TF-1100_1.4.0-2-5 | Mailbox Read Service Repeat 2 | Error
TF-1100_1.4.0-2-5 | Mailbox Read Service Repeat 2 | Error
TF-1100_1.4.0-5-1 | Check Device Type (Complex or Simple Type) | Success
TF-1100_1.4.0-5-2 | Secondary Vendor ID in IP Core | Success
TF-1200_1.5.0-1-1 | #1.1 Init > Init, ErrFlag = 1, AckFlag = 0, AlCtrl = 0x01 | Success
TF-1200_1.5.0-1-2 | #1.2A Init > Init, ErrFlag = 1, AckFlag = 0, AlCtrl = 0x02 | Success
TF-1200_1.5.0-1-3 | #1.2B Init > Init, ErrFlag = 1, AckFlag = 0, AlCtrl= 0x03 | Success
TF-1200_1.5.0-1-4 | #1.2C Init > Init, ErrFlag = 1, AckFlag = 0, AlCtrl= 0x04 | Success
TF-1200_1.5.0-1-5 | #1.2D Init > Init, ErrFlag = 1, AckFlag = 0, AlCtrl= 0x08 | Success
TF-1200_1.5.0-1-6 | #1.2E Init > Init, ErrFlag = 1, AckFlag = 0, AlCtrl= unknown | Success
TF-1200_1.5.0-1-7 | #2 Init > Init, ErrFlag = 0 or AckFlag = 1, AlCtrl = 0x01 | Success
TF-1200_1.5.0-1-7 | #2 Init > Init, ErrFlag = 0 or AckFlag = 1, AlCtrl = 0x01 | Success
TF-1200_1.5.0-1-8 | #3 Init > PreOp, ErrFlag = 0 or AckFlag = 1, AlCtrl = 0x02, SmSett01 match | Success
TF-1200_1.5.0-1-8 | #3 Init > PreOp, ErrFlag = 0 or AckFlag = 1, AlCtrl = 0x02, SmSett01 match | Success
TF-1200_1.5.0-1-9 | #4 Init > Init,  ErrFlag = 0 or AckFlag = 1, AlCtrl = 0x02, SmSett0 or SmSett1 not match | Success
TF-1200_1.5.0-1-9 | #4 Init > Init,  ErrFlag = 0 or AckFlag = 1, AlCtrl = 0x02, SmSett0 or SmSett1 not match | Success
TF-1200_1.5.0-1-9 | #4 Init > Init,  ErrFlag = 0 or AckFlag = 1, AlCtrl = 0x02, SmSett0 or SmSett1 not match | Success
TF-1200_1.5.0-1-9 | #4 Init > Init,  ErrFlag = 0 or AckFlag = 1, AlCtrl = 0x02, SmSett0 or SmSett1 not match | Success
TF-1200_1.5.0-1-12 | # 7 Init > Init, ErrFlag = 0 or AckFlag = 1, AlCtrl  = 0x03, Boot not supported | Success
TF-1200_1.5.0-1-12 | # 7 Init > Init, ErrFlag = 0 or AckFlag = 1, AlCtrl  = 0x03, Boot not supported | Success
TF-1200_1.5.0-1-13 | # 8 Init > Init, ErrFlag = 0 or AckFlag = 1, AlCtrl = 0x04 or 0x08 | Success
TF-1200_1.5.0-1-13 | # 8 Init > Init, ErrFlag = 0 or AckFlag = 1, AlCtrl = 0x04 or 0x08 | Success
TF-1200_1.5.0-1-13 | # 8 Init > Init, ErrFlag = 0 or AckFlag = 1, AlCtrl = 0x04 or 0x08 | Success
TF-1200_1.5.0-1-13 | # 8 Init > Init, ErrFlag = 0 or AckFlag = 1, AlCtrl = 0x04 or 0x08 | Success
TF-1200_1.5.0-1-14 | # 9 Init > Init, ErrFlag = 0 or AckFlag = 1, AlCtrl = unknown | Success
TF-1200_1.5.0-1-14 | # 9 Init > Init, ErrFlag = 0 or AckFlag = 1, AlCtrl = unknown | Success
TF-1200_1.5.0-1-15 | #10 Init > Init, Sm_Change ignored | Success
TF-1200_1.5.0-1-15 | #10 Init > Init, Sm_Change ignored | Success
TF-1200_1.5.0-1-15 | #10 Init > Init, Sm_Change ignored | Success
TF-1200_1.5.0-1-15 | #10 Init > Init, Sm_Change ignored | Success
TF-1200_1.5.0-2-1 | #11.1 PreOp > Init, ErrFlag = 1, AckFlag = 0, AlCtrl = 0x01 | Success
TF-1200_1.5.0-2-2 | #11.2A PreOp > PreOp, ErrFlag = 1, AckFlag = 0, AlCtrl = 0x02 | Success
TF-1200_1.5.0-2-3 | #11.2B PreOp > PreOp, ErrFlag = 1, AckFlag = 0, AlCtrl = 0x03 | Success
TF-1200_1.5.0-2-4 | #11.2C PreOp > PreOp, ErrFlag = 1, AckFlag = 0, AlCtrl = 0x04 | Success
TF-1200_1.5.0-2-5 | #11.2D PreOp > PreOp, ErrFlag = 1, AckFlag = 0, AlCtrl = 0x08 | Success
TF-1200_1.5.0-2-6 | #11.2E PreOp > PreOp, ErrFlag = 1, AckFlag = 0, AlCtrl = unknown | Success
TF-1200_1.5.0-2-7 | #12 PreOp > Init, ErrFlag = 0 or AckFlag = 1, AlCtrl = 0x01 | Success
TF-1200_1.5.0-2-7 | #12 PreOp > Init, ErrFlag = 0 or AckFlag = 1, AlCtrl = 0x01 | Success
TF-1200_1.5.0-2-8 | #13 PreOp > PreOp, ErrFlag = 0 or AckFlag = 1, AlCtrl = 0x02 | Success
TF-1200_1.5.0-2-8 | #13 PreOp > PreOp, ErrFlag = 0 or AckFlag = 1, AlCtrl = 0x02 | Success
TF-1200_1.5.0-2-9 | #17  PreOp > PreOp, ErrFlag = 0 or ErrAck = 1, AlCtrl = 0x04, SmSett_2_To_n not match | Success
TF-1200_1.5.0-2-9 | #17  PreOp > PreOp, ErrFlag = 0 or ErrAck = 1, AlCtrl = 0x04, SmSett_2_To_n not match | Success
TF-1200_1.5.0-2-9 | #17  PreOp > PreOp, ErrFlag = 0 or ErrAck = 1, AlCtrl = 0x04, SmSett_2_To_n not match | Success
TF-1200_1.5.0-2-10 | # 18 PreOp > PreOp,  ErrFlag = 0 or AckFlag = 1, AlCtrl = 0x03 or 0x08 | Success
TF-1200_1.5.0-2-10 | # 18 PreOp > PreOp,  ErrFlag = 0 or AckFlag = 1, AlCtrl = 0x03 or 0x08 | Success
TF-1200_1.5.0-2-10 | # 18 PreOp > PreOp,  ErrFlag = 0 or AckFlag = 1, AlCtrl = 0x03 or 0x08 | Success
TF-1200_1.5.0-2-10 | # 18 PreOp > PreOp,  ErrFlag = 0 or AckFlag = 1, AlCtrl = 0x03 or 0x08 | Success
TF-1200_1.5.0-2-11 | # 19 PreOp > PreOp,  ErrFlag = 0 or AckFlag = 1, AlCtrl = unknown | Success
TF-1200_1.5.0-2-11 | # 19 PreOp > PreOp,  ErrFlag = 0 or AckFlag = 1, AlCtrl = unknown | Success
TF-1200_1.5.0-2-13 | # 20.2  PreOp > PreOp, Sm_Chg, ErrFlag = 1 | Success
TF-1200_1.5.0-2-13 | # 20.2  PreOp > PreOp, Sm_Chg, ErrFlag = 1 | Success
TF-1200_1.5.0-2-13 | # 20.2  PreOp > PreOp, Sm_Chg, ErrFlag = 1 | Success
TF-1200_1.5.0-2-13 | # 20.2  PreOp > PreOp, Sm_Chg, ErrFlag = 1 | Success
TF-1200_1.5.0-2-14 | # 21 PreOp > Init, Sm_Chg, ErrFlag = 0, SmSett0 or SmSett1 not match | Success
TF-1200_1.5.0-2-14 | # 21 PreOp > Init, Sm_Chg, ErrFlag = 0, SmSett0 or SmSett1 not match | Success
TF-1200_1.5.0-3-1 | #22.1 SafeOp > Init, ErrFlag  = 1, AckFlag = 0, AlCtrl = 0x01 | Success
TF-1200_1.5.0-3-2 | #22.2A SafeOp > SafeOp, ErrFlag = 1, AckFlag = 0, AlCtrl = 0x02 | Success
TF-1200_1.5.0-3-3 | #22.2B SafeOp > SafeOp, ErrFlag = 1, AckFlag = 0, AlCtrl = 0x03 | Success
TF-1200_1.5.0-3-4 | #22.2C SafeOp > SafeOp, ErrFlag = 1, AckFlag = 0, AlCtrl = 0x04 | Success
TF-1200_1.5.0-3-5 | #22.2D SafeOp > SafeOp, ErrFlag = 1, AckFlag = 0, AlCtrl = 0x08 | Success
TF-1200_1.5.0-3-6 | #22.2E SafeOp > SafeOp, ErrFlag = 1, AckFlag = 0, AlCtrl = unknown | Success
TF-1200_1.5.0-3-7 | #23 SafeOp > Init, ErrFlag = 0 or AckFlag = 1, AlCtrl = 0x01 | Success
TF-1200_1.5.0-3-7 | #23 SafeOp > Init, ErrFlag = 0 or AckFlag = 1, AlCtrl = 0x01 | Success
TF-1200_1.5.0-3-8 | #24 SafeOp > PreOp, ErrFlag = 0 or ErrAck = 1,  AlCtrl = 0x02 | Success
TF-1200_1.5.0-3-8 | #24 SafeOp > PreOp, ErrFlag = 0 or ErrAck = 1,  AlCtrl = 0x02 | Success
TF-1200_1.5.0-3-9 | #25.1 SafeOp > SafeOp, ErrFlag = 1, localErrFlag = 0, AckFlag = 1, AlCtrl = 0x04 | Success
TF-1200_1.5.0-3-11 | #29 SafeOp > SafeOp, AL ErrFlag = 0 or AckFlag = 1, AlCtrl = 0x03 | Success
TF-1200_1.5.0-3-11 | #29 SafeOp > SafeOp, AL ErrFlag = 0 or AckFlag = 1, AlCtrl = 0x03 | Success
TF-1200_1.5.0-3-12 | #30 SafeOp > SafeOp, ErrFlag = 0 or AckFlag = 1, AlCtrl = unknown | Success
TF-1200_1.5.0-3-12 | #30 SafeOp > SafeOp, ErrFlag = 0 or AckFlag = 1, AlCtrl = unknown | Success
TF-1200_1.5.0-3-14 | #31.2 SafeOp > SafeOp, ErrFlag = 1, Sm_Chg | Success
TF-1200_1.5.0-3-14 | #31.2 SafeOp > SafeOp, ErrFlag = 1, Sm_Chg | Success
TF-1200_1.5.0-3-14 | #31.2 SafeOp > SafeOp, ErrFlag = 1, Sm_Chg | Success
TF-1200_1.5.0-3-14 | #31.2 SafeOp > SafeOp, ErrFlag = 1, Sm_Chg | Success
TF-1200_1.5.0-3-15 | #32 SafeOp > PreOp Sm_Chg, ErrFlag = 0, SmSett01 match, SmSett 23 not match | Success
TF-1200_1.5.0-3-15 | #32 SafeOp > PreOp Sm_Chg, ErrFlag = 0, SmSett01 match, SmSett 23 not match | Success
TF-1200_1.5.0-3-16 | #33 SafeOp > Init, ErrFlag = 0, SmSett0 or SmSett1 not match | Success
TF-1200_1.5.0-3-16 | #33 SafeOp > Init, ErrFlag = 0, SmSett0 or SmSett1 not match | Success
TF-1200_1.5.0-4-1 | #37 Op > Init, Ack = 0 or 1, AlCtrl = 0x01 | Success
TF-1200_1.5.0-4-1 | #37 Op > Init, Ack = 0 or 1, AlCtrl = 0x01 | Success
TF-1200_1.5.0-4-2 | #38 Op > PreOp, AckFlag = 0 or 1, AlCtrl = 0x02 | Success
TF-1200_1.5.0-4-2 | #38 Op > PreOp, AckFlag = 0 or 1, AlCtrl = 0x02 | Success
TF-1200_1.5.0-4-3 | #39 Op > SafeOp, AckFlag = 0 or 1, AlCtrl = 0x04 | Success
TF-1200_1.5.0-4-3 | #39 Op > SafeOp, AckFlag = 0 or 1, AlCtrl = 0x04 | Success
TF-1200_1.5.0-4-4 | #40 Op > Op, AckFlag 0 or 1, AlCtrl = 0x08 | Success
TF-1200_1.5.0-4-4 | #40 Op > Op, AckFlag 0 or 1, AlCtrl = 0x08 | Success
TF-1200_1.5.0-4-5 | #42 Op > SafeOp, AckFlag = 0 or 1, AlCtrl = 0x03 | Success
TF-1200_1.5.0-4-5 | #42 Op > SafeOp, AckFlag = 0 or 1, AlCtrl = 0x03 | Success
TF-1200_1.5.0-4-6 | #43 Op > SafeOp, ErrFlag = 0 or AckFlag = 1, AlCtrl = unknown | Success
TF-1200_1.5.0-4-6 | #43 Op > SafeOp, ErrFlag = 0 or AckFlag = 1, AlCtrl = unknown | Success
TF-1200_1.5.0-4-8 | #45  Op > PreOp, Sm_Chg, ErrFlag = 0, SmSett01 match, SmSett 23 not match | Success
TF-1200_1.5.0-4-8 | #45  Op > PreOp, Sm_Chg, ErrFlag = 0, SmSett01 match, SmSett 23 not match | Success
TF-1200_1.5.0-4-9 | #46 Op > Init, SmSett0 or SmSett1 not match | Success
TF-1200_1.5.0-4-9 | #46 Op > Init, SmSett0 or SmSett1 not match | Success
TF-1200_1.5.0-4-10 | #49 Op > SafeOp, WD expired | Success
TF-1201_1.5.0-5-5 | Ignore ID Request if IdentificationReg134 = false in INIT | Success
TF-1201_1.5.0-5-5 | Ignore ID Request if IdentificationReg134 = false in INIT | Success
TF-1201_1.5.0-5-6 | Ignore ID Request if IdentificationReg134 = false in PREOP | Success
TF-1201_1.5.0-5-6 | Ignore ID Request if IdentificationReg134 = false in PREOP | Success
TF-1201_1.5.0-5-7 | Ignore ID Request if IdentificationReg134 = false in SAFEOP | Success
TF-1201_1.5.0-5-7 | Ignore ID Request if IdentificationReg134 = false in SAFEOP | Success
TF-1201_1.5.0-5-8 | Ignore ID Request if IdentificationReg134 = false in OP | Success
TF-1201_1.5.0-5-10 | ConfiguredStationAlias value | Success
TF-2300_1.6.0-1-1 | Sync Manager PDO Assignment (online) | SeriousError
TF-2300_1.6.0-1-1 | Sync Manager PDO Assignment (online) | Error
TF-2300_1.6.0-1-2 | Receive and Transmit PDO Mapping (online) | SeriousError
TF-2300_1.6.0-1-2 | Receive and Transmit PDO Mapping (online) | Error
TF-2300_1.6.0-1-3 | Download RO in PREOP SDOs (online) | SeriousError
TF-2300_1.6.0-1-3 | Download RO in PREOP SDOs (online) | Error
TF-2300_1.6.0-1-4 | Download RO in SAFEOP SDOs (online) | SeriousError
TF-2300_1.6.0-1-4 | Download RO in SAFEOP SDOs (online) | Error
TF-2300_1.6.0-1-5 | Download RO in OP SDOs (online) | SeriousError
TF-2300_1.6.0-1-5 | Download RO in OP SDOs (online) | Error
TF-2300_1.6.0-1-6 | Download WO in PREOP SDOs (online) | SeriousError
TF-2300_1.6.0-1-6 | Download WO in PREOP SDOs (online) | Error
TF-2300_1.6.0-1-7 | Download WO in SAFEOP SDOs (online) | SeriousError
TF-2300_1.6.0-1-7 | Download WO in SAFEOP SDOs (online) | Error
TF-2300_1.6.0-1-8 | Download WO in OP SDOs (online) | SeriousError
TF-2300_1.6.0-1-8 | Download WO in OP SDOs (online) | Error
TF-2300_1.6.0-1-9 | Download RW in PREOP SDOs (online) | SeriousError
TF-2300_1.6.0-1-9 | Download RW in PREOP SDOs (online) | Error
TF-2300_1.6.0-1-10 | Download RW in SAFEOP SDOs (online) | SeriousError
TF-2300_1.6.0-1-10 | Download RW in SAFEOP SDOs (online) | Error
TF-2300_1.6.0-1-11 | Download RW in OP SDOs (online) | SeriousError
TF-2300_1.6.0-1-11 | Download RW in OP SDOs (online) | Error
TF-2300_1.6.0-1-12 | Upload WO in PREOP SDOs (online) | SeriousError
TF-2300_1.6.0-1-12 | Upload WO in PREOP SDOs (online) | Error
TF-2300_1.6.0-1-13 | Upload WO in SAFEOP SDOs (online) | SeriousError
TF-2300_1.6.0-1-13 | Upload WO in SAFEOP SDOs (online) | Error
TF-2300_1.6.0-1-14 | Upload WO in OP SDOs (online) | SeriousError
TF-2300_1.6.0-1-14 | Upload WO in OP SDOs (online) | Error
TF-2300_1.6.0-1-15 | Upload RW in PREOP SDOs (online) | SeriousError
TF-2300_1.6.0-1-15 | Upload RW in PREOP SDOs (online) | Error
TF-2300_1.6.0-1-16 | Upload RW in SAFEOP SDOs (online) | SeriousError
TF-2300_1.6.0-1-16 | Upload RW in SAFEOP SDOs (online) | Error
TF-2300_1.6.0-1-17 | Upload RW in OP SDOs (online) | SeriousError
TF-2300_1.6.0-1-17 | Upload RW in OP SDOs (online) | Error
TF-2300_1.6.0-1-18 | Upload RO in PREOP SDOs (online) | SeriousError
TF-2300_1.6.0-1-18 | Upload RO in PREOP SDOs (online) | Error
TF-2300_1.6.0-1-19 | Upload RO in SAFEOP SDOs (online) | SeriousError
TF-2300_1.6.0-1-19 | Upload RO in SAFEOP SDOs (online) | Error
TF-2300_1.6.0-1-20 | Upload RO in OP SDOs (online) | SeriousError
TF-2300_1.6.0-1-20 | Upload RO in OP SDOs (online) | Error
TF-2300_1.6.0-2-1 | Sync Manager PDO Assignment (offline) | Success
TF-2300_1.6.0-2-2 | Receive and Transmit PDO Mapping (offline) | Success
TF-2300_1.6.0-2-3 | Download RO in PREOP SDOs (offline) | Error
TF-2300_1.6.0-2-4 | Download RO in SAFEOP SDOs (offline) | Error
TF-2300_1.6.0-2-5 | Download RO in OP SDOs (offline) | Error
TF-2300_1.6.0-2-9 | Download RW in PREOP SDOs (offline) | SeriousError
TF-2300_1.6.0-2-9 | Download RW in PREOP SDOs (offline) | Error
TF-2301_1.5.0-1-1 | Compare online to offline object dictionary | SeriousError
TF-2301_1.5.0-1-1 | Compare online to offline object dictionary | Error
TF-2301_1.5.0-3-1 | Device Type Object (0x1000) | Error
TF-2301_1.5.0-3-1 | Device Type Object (0x1000) | Error
TF-2301_1.5.0-3-6 | Identity Object (0x1018) | Success
TF-2301_1.5.0-3-8 | Diagnosis History Object (0x10F3) | Success
TF-2301_1.5.0-3-13 | Sync Manager PDO Assignment (0x1C10 .. 0x1C2F) | Warning
TF-2301_1.5.0-3-13 | Sync Manager PDO Assignment (0x1C10 .. 0x1C2F) | Warning
TF-2301_1.5.0-3-13 | Sync Manager PDO Assignment (0x1C10 .. 0x1C2F) | Success
TF-2301_1.5.0-3-14 | Sync Manager Parameter (0x1C30 - 0x1C4F) | Success
TF-2301_1.5.0-5-1 | Device Type Object (0x1000) | SeriousError
TF-2301_1.5.0-5-1 | Device Type Object (0x1000) | Error
TF-2301_1.5.0-5-2 | Error Register Object (0x1001) | SeriousError
TF-2301_1.5.0-5-2 | Error Register Object (0x1001) | Error
TF-2301_1.5.0-5-3 | Manufacturer Device Name Object (0x1008) | SeriousError
TF-2301_1.5.0-5-3 | Manufacturer Device Name Object (0x1008) | Error
TF-2301_1.5.0-5-4 | Manufacturer Hardware Version Object (0x1009) | SeriousError
TF-2301_1.5.0-5-4 | Manufacturer Hardware Version Object (0x1009) | Error
TF-2301_1.5.0-5-5 | Manufacturer Software Version Object (0x100A) | SeriousError
TF-2301_1.5.0-5-5 | Manufacturer Software Version Object (0x100A) | Error
TF-2301_1.5.0-5-6 | Identity Object (0x1018) | SeriousError
TF-2301_1.5.0-5-6 | Identity Object (0x1018) | Error
TF-2301_1.5.0-5-7 | Error Setting Object (0x10F1) | SeriousError
TF-2301_1.5.0-5-7 | Error Setting Object (0x10F1) | Error
TF-2301_1.5.0-5-8 | Diagnosis History Object (0x10F3) | SeriousError
TF-2301_1.5.0-5-8 | Diagnosis History Object (0x10F3) | Error
TF-2301_1.5.0-5-9 | RxPDO Parameter (0x1400 .. 0x15FF) | SeriousError
TF-2301_1.5.0-5-9 | RxPDO Parameter (0x1400 .. 0x15FF) | Error
TF-2301_1.5.0-5-10 | TxPDO Parameter (0x1800 .. 0x19FF) | SeriousError
TF-2301_1.5.0-5-10 | TxPDO Parameter (0x1800 .. 0x19FF) | Error
TF-2301_1.5.0-5-12 | Sync Manager Communication Type Object (0x1C00) | SeriousError
TF-2301_1.5.0-5-12 | Sync Manager Communication Type Object (0x1C00) | Error
TF-2301_1.5.0-5-13 | Sync Manager PDO Assignment (0x1C10 .. 0x1C2F) | SeriousError
TF-2301_1.5.0-5-13 | Sync Manager PDO Assignment (0x1C10 .. 0x1C2F) | Error
TF-2301_1.5.0-5-14 | Sync Manager Parameter (0x1C30 - 0x1C4F) | SeriousError
TF-2301_1.5.0-5-14 | Sync Manager Parameter (0x1C30 - 0x1C4F) | Error
TF-2301_1.5.0-6-10 | Device Command (0xFB00 .. 0xFBFF) | SeriousError
TF-2301_1.5.0-6-10 | Device Command (0xFB00 .. 0xFBFF) | Error
TF-2302_1.3.0-1-1 | Objects with read Access in PreOp (offline dictionary) | Error
TF-2302_1.3.0-1-2 | Objects with read Access in SafeOp (offline dictionary) | Error
TF-2302_1.3.0-1-3 | Objects with read Access in Op (offline dictionary) | Error
TF-2302_1.3.0-2-1 | Object Access RW in PreOp (offline dictionary) | Error
TF-2302_1.3.0-2-2 | Object Access RW in SafeOp (offline dictionary) | Error
TF-2302_1.3.0-2-3 | Object Access RW in Op (offline dictionary) | Error
TF-2302_1.3.0-3-1 | Objects with read Access in PreOp (online dictionary) | SeriousError
TF-2302_1.3.0-3-1 | Objects with read Access in PreOp (online dictionary) | Error
TF-2302_1.3.0-3-2 | Objects with read Access in SafeOp (online dictionary) | SeriousError
TF-2302_1.3.0-3-2 | Objects with read Access in SafeOp (online dictionary) | Error
TF-2302_1.3.0-3-3 | Objects with read Access in Op (online dictionary) | SeriousError
TF-2302_1.3.0-3-3 | Objects with read Access in Op (online dictionary) | Error
TF-2302_1.3.0-4-1 | Object Access RW in PreOp (online dictionary) | SeriousError
TF-2302_1.3.0-4-1 | Object Access RW in PreOp (online dictionary) | Error
TF-2302_1.3.0-4-2 | Object Access RW in SafeOp (online dictionary) | SeriousError
TF-2302_1.3.0-4-2 | Object Access RW in SafeOp (online dictionary) | Error
TF-2302_1.3.0-4-3 | Object Access RW in Op (online dictionary) | SeriousError
TF-2302_1.3.0-4-3 | Object Access RW in Op (online dictionary) | Error
TF-4101_1.2.0-1-1 | offline \| 0x1000 | Error
TF-4101_1.2.0-1-3 | offline \| 0x1001 | Error
TF-4101_1.2.0-2-1 | offline \| 0x6000 to 0x6006 (reserved range) | Warning

</body>

</html>

### CODE REVIEW

File: lib/slave/include/kickcat/AbstractESC.h [hunk 1]
```diff
@@ -26,8 +26,8 @@ namespace kickcat
     #define SYNC_MANAGER_PI_IN(index, address, length)  SyncManagerConfig{index, address, length, 0x20, SyncManagerType::Input}
     #define SYNC_MANAGER_PI_OUT(index, address, length) SyncManagerConfig{index, address, length, 0x64, SyncManagerType::Output}
 
-    #define SYNC_MANAGER_MBX_IN(index, address, length)  SyncManagerConfig{index, address, length, 0x02, SyncManagerType::MailboxIn}
-    #define SYNC_MANAGER_MBX_OUT(index, address, length) SyncManagerConfig{index, address, length, 0x06, SyncManagerType::MailboxOut}
+    #define SYNC_MANAGER_MBX_IN(index, address, length)  SyncManagerConfig{index, address, length, 0x22, SyncManagerType::MailboxIn}
```
Comment (nicolas71640) [inline lib/slave/include/kickcat/AbstractESC.h:29]: @leducp  Does this change ok ? This allow the config and the sii to match

File: lib/slave/src/AbstractESC.cc [hunk 2]
```diff
@@ -305,6 +328,23 @@ namespace kickcat
 
         update_process_data_input();
         update_process_data_output();
+        
+        if (not are_valid_sm(sm_mailbox_configs_))
```
Comment (Rdk-T) [inline lib/slave/src/AbstractESC.cc:332]: Is this check really supposed to be in op ? it's strand that it's check each time and then go to init while putting preop in error diagnostic

Comment (nicolas71640) [reply lib/slave/src/AbstractESC.cc:332]: Yep... it is. This test need it : 'Op > Init, SmSett0 or SmSett1 not match'

Comment (nicolas71640) [reply lib/slave/src/AbstractESC.cc:332]: It's seams quite brutal though (even though we did the same for the pdo sm). 
Maybe we could do something with an interruption. I'll open a issue to follow this matter

Comment (nicolas71640) [reply lib/slave/src/AbstractESC.cc:332]: Here's the issue

File: lib/slave/include/kickcat/AbstractESC.h (hunk 1 above)
Comment (nicolas71640) [reply lib/slave/include/kickcat/AbstractESC.h]: Removed this change and just testing operation mode and direction in the is_valid_sm()

**Review by Rdk-T — APPROVED** [review]

**Review by leducp — APPROVED** [review]

**Pull request merged**